esphome/components/wifi_info/text_sensor.py

```diff
@@ -71,16 +80,7 @@ async def setup_conf(config, key):
 
 async def to_code(config):
     # Request WiFi callbacks for any sensor that needs them
-    if any(
-        key in config
-        for key in (
-            CONF_SSID,
-            CONF_BSSID,
-            CONF_IP_ADDRESS,
-            CONF_DNS_ADDRESS,
-            CONF_SCAN_RESULTS,
-        )
-    ):
+    if _NETWORK_INFO_KEYS & config.keys():
```

Comment:
I think you can drop .keys() since keys is default

Reply:
or `_NETWORK_INFO_KEYS.intersection(config)` -- but & is fine as well I think its the same

Reply:
I tried but `TypeError: unsupported operand type(s) for &: 'set' and 'OrderedDict'`

Reply:
This seems to work -- maybe it's better?
```suggestion
    if _NETWORK_INFO_KEYS.intersection(config):
```